File: src/main/java/io/unthrottled/amii/listeners/IDEPluginInstallListener.java
package io.unthrottled.amii.listeners;

import com.intellij.ide.plugins.DynamicPluginListener;
import com.intellij.ide.plugins.IdeaPluginDescriptor;
import com.intellij.openapi.application.ApplicationManager;
import io.unthrottled.amii.PluginMaster;

import static io.unthrottled.amii.config.Constants.PLUGIN_ID;

public class IDEPluginInstallListener implements DynamicPluginListener {
  @Override
  public void pluginLoaded(IdeaPluginDescriptor pluginDescriptor) {
    if (PLUGIN_ID.equals(pluginDescriptor.getPluginId().getIdString())) {
      ApplicationManager.getApplication().invokeLater(() -> PluginMaster.Companion.getInstance().onUpdate());
    }
  }
}

File: src/main/kotlin/io/unthrottled/amii/listeners/IDEPluginInstallListener.kt
package io.unthrottled.amii.listeners

import com.intellij.ide.plugins.DynamicPluginListener
import com.intellij.ide.plugins.IdeaPluginDescriptor
import com.intellij.openapi.application.ApplicationManager
import io.unthrottled.amii.PluginMaster
import io.unthrottled.amii.config.Constants.PLUGIN_ID
import io.unthrottled.amii.tools.Logging

class IDEPluginInstallListener : DynamicPluginListener, Logging {

  override fun beforePluginLoaded(pluginDescriptor: IdeaPluginDescriptor) {
  }

  override fun beforePluginUnload(
    pluginDescriptor: IdeaPluginDescriptor,
    isUpdate: Boolean
  ) {
  }

  override fun checkUnloadPlugin(pluginDescriptor: IdeaPluginDescriptor) {
  }

  override fun pluginLoaded(pluginDescriptor: IdeaPluginDescriptor) {
    if (pluginDescriptor.pluginId.idString == PLUGIN_ID) {
      ApplicationManager.getApplication().invokeLater {
        PluginMaster.instance.onUpdate()
      }
    }
  }

  override fun pluginUnloaded(
    pluginDescriptor: IdeaPluginDescriptor,
    isUpdate: Boolean
  ) {
  }
}
